intranet/apps/preferences/tests.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+from django.core.urlresolvers import reverse
+from ...test.ion_test import IonTestCase
+from mock import patch
+from . import views
+
+
+class PreferencesTest(IonTestCase):
+
+    def test_get_preferences(self):
+        self.login()
+        response = self.client.get(reverse('preferences'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_set_preferences(self):
+        self.login()
+        response = self.client.post(reverse('preferences'),
+                                    {'showpictures-self': ['on'],
+                                     'mobile_phone': ['5555555555'],
+                                     'showschedule-self': ['on'],
+                                     'home_phone': ['5555555555'],
+                                     'webpage_0': ['https://ion.tjhsst.edu/logout'],
+                                     'receive_push_notifications': ['on'],
+                                     'email_0': ['awilliam@tjhsst.edu'],
+                                     'showbirthday-self': ['on'],
+                                     'preferred_photo': ['AUTO'],
+                                     'other_phone_0': ['5555555555']})
+        self.assertEqual(response.status_code, 302)
+        response = self.client.post(reverse('preferences'), "{}")
+        self.assertEqual(response.status_code, 404)
+
+    @patch('views.logger')
```

Comment:
Don't mock out the logger, use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs